testing/test_nodes.py

```diff
@@ -22,6 +22,18 @@ def test_ischildnode(baseid, nodeid, expected):
     assert result is expected
 
 
+def test_direct_ctor_warning():
```

Comment:
I think this test should go to `deprecated_test.py`

Reply:
Also, please make sure the stack level is correct (https://github.com/pytest-dev/pytest/pull/5984).